hummingbot/connector/exchange/gate_io/gate_io_exchange.py

```diff
@@ -782,14 +779,15 @@ async def _process_trade_message(self, trade_msg: Dict[str, Any]):
             "price": "10000.00000000",
             "fee": "0.00200000000000",
             "point_fee": "0",
-            "gt_fee": "0"
+            "gt_fee": "0",
+            "text": "user_defined_text",
         }
         """
         exchange_order_id = str(trade_msg["order_id"])
+        client_order_id = str(trade_msg["text"])
         tracked_orders = list(self._in_flight_orders.values())
-        for order in tracked_orders:
-            await order.get_exchange_order_id()
-        track_order = [o for o in tracked_orders if exchange_order_id == o.exchange_order_id]
+        track_order = [o for o in tracked_orders
+                       if exchange_order_id == o.exchange_order_id or client_order_id == o.client_order_id]
```

Comment:
Since `_in_flight_orders` is a mapping of `client_order_id` to the `GateIoInFlightOrder`, it would be simpler to just do the following

```python
client_order_id = str(trade_msg["text"])
if client_order_id not in self._in_flight_orders:
    return
tracked_order = self._in_flight_orders[client_order_id]
```

Reply:
The Gateio documentation is not very clear on weather the `text` component of the message always includes the client order id (in fact, the documentation does not mention it. I realized the field contained the client order id doing manual tests). I don't feel comfortable shifting the main logic from considering more important the exchange order id.

Reply:
You're right, their API docs does not explicitly mention that the `text` field will be returned every time, but it appears that the `text` field does refer to the `client_order_id`. I also noticed that we have a prefix of `t-HBOT` for all our gateio client order id.

![image](https://user-images.githubusercontent.com/6755922/135462745-4e86a61e-9beb-4486-9285-5eec19d01c2e.png)